cmd/akamai-purger/main.go

```diff
@@ -24,40 +23,150 @@ import (
+	akamaiAPIReqPerSecondLimit = 50
+
+	// akamaiAPIReqPerSecondLimit is the limit of URLs, sent per second, that
+	// we're allowed to make to the Fast-Purge API.
+	akamaiURLsPerSecondLimit = 200
+)
+
+// Throughput is a container for all throuput related akamai-purger
+// configuration settings.
+type Throughput struct {
+
+	// ResponsesPerBatch the number of cached OCSP responses to included in each
+	// purge request. One cached OCSP response is composed of 3 URLs totaling <
+	// 400 bytes. If this value isn't provided it will default to
+	// 'defaultResponsesPerBatch'.
+	ResponsesPerBatch int
+
+	// PurgeBatchEvery is the duration waited between dispatching an Akamai
+	// purge request containing 'ResponsesPerBatch' * 3 URLs. If this value
+	// isn't provided it will default to 'defaultPurgeBatchEvery'.
+	PurgeBatchEvery cmd.ConfigDuration
+}
+
+func (t *Throughput) useOptimizedDefaults() {
+	if t.ResponsesPerBatch == 0 {
+		t.ResponsesPerBatch = defaultResponsesPerBatch
+	}
+	if t.PurgeBatchEvery.Duration == 0 {
+		t.PurgeBatchEvery.Duration = defaultPurgeBatchEvery
+	}
+}
+
+// validate ensures that the provided throughput configuration will not violate
+// the Akamai Fast-Purge API limits. For more information see the official
+// documentation:
+// https://techdocs.akamai.com/purge-cache/reference/rate-limiting
+func (t *Throughput) validate() error {
+	if t.PurgeBatchEvery.Duration == 0 {
+		// TODO(#6003) remove /'purgeInterval'.
+		return errors.New("'purgeBatchEvery'/'purgeInterval' must be > 0 nanoseconds")
+	}
+	if t.ResponsesPerBatch <= 0 {
+		return errors.New("'responsesPerBatch' must be > 0")
+	}
+
+	// Send no more than the 50,000 bytes of objects we’re allotted per request.
+	bytesPerRequest := (t.ResponsesPerBatch * akamaiBytesPerResponse)
+	if bytesPerRequest > akamaiBytesPerReqLimit {
+		return fmt.Errorf("configuration exceeds Akamai's bytes per request limit (%d bytes) by %d",
+			akamaiBytesPerReqLimit, bytesPerRequest-akamaiBytesPerReqLimit)
+	}
+
+	// Send no more than the 50 API requests we’re allotted each second.
+	requestsPerSecond := int(time.Second / t.PurgeBatchEvery.Duration)
+	if requestsPerSecond > akamaiAPIReqPerSecondLimit {
+		return fmt.Errorf("configuration exceeds Akamai's requests per second limit (%d requests) by %d",
+			akamaiAPIReqPerSecondLimit, requestsPerSecond-akamaiAPIReqPerSecondLimit)
+	}
+
+	// Purge no more than the 200 URLs we’re allotted each second.
+	urlsPurgedPerSecond := requestsPerSecond * (t.ResponsesPerBatch * 3)
+	if urlsPurgedPerSecond > akamaiURLsPerSecondLimit {
+		return fmt.Errorf("configuration exceeds Akamai's URLs purged per second limit (%d URLs) by %d",
+			akamaiURLsPerSecondLimit, urlsPurgedPerSecond-akamaiURLsPerSecondLimit)
+	}
+	return nil
+}
 
 type Config struct {
 	AkamaiPurger struct {
 		cmd.ServiceConfig
 
-		// PurgeInterval is the duration waited between purge requests.
+		// PurgeInterval is DEPRECATED in favor of the 'PurgeBatchEvery' field
```

Comment:
If you mark the deprecation by starting a separate line/paragraph of the doc-comment with "Deprecated:", then linters can pick up on it automatically.